src/main.c

```diff
@@ -464,6 +466,49 @@ static int listener_setup_ssl(h2o_configurator_command_t *cmd, h2o_configurator_
                 }
                 continue;
             }
+            if (strcmp(key->data.scalar, "http2-origin-frame") == 0) {
+                switch (value->type) {
+                case YOML_TYPE_SCALAR:
+                {
+                    uint16_t name_size;
+                    name_size = strlen(value->data.scalar);
+                    h2o_strtolower(value->data.scalar, name_size);
+                    http2_origin_frame = h2o_mem_alloc(sizeof(*http2_origin_frame));
+                    http2_origin_frame->len = sizeof(uint16_t) + name_size;
+                    http2_origin_frame->base = h2o_mem_alloc(http2_origin_frame->len);
+
+                    memcpy(http2_origin_frame->base + sizeof(name_size), value->data.scalar, name_size);
+                    name_size = htons(name_size);
+                    memcpy(http2_origin_frame->base, &name_size, sizeof(name_size));
+                } break;
+                case YOML_TYPE_SEQUENCE:
+                {
+                     size_t i;
+                     uint16_t elem_lens[value->data.sequence.size];
+                     h2o_iovec_t elems[value->data.sequence.size * 2];
+                     for (i = 0; i != value->data.sequence.size; ++i) {
+                         yoml_t *element = value->data.sequence.elements[i];
+                         if (element->type != YOML_TYPE_SCALAR) {
+                             h2o_configurator_errprintf(cmd, element, "element of a sequence passed to unsetenv must be a scalar");
+                             return -1;
+                         }
+                         elem_lens[i] = htons(strlen(element->data.scalar));
+                         elems[i*2].base = (char *)&elem_lens[i];
+                         elems[i*2].len = 2;
+                         elems[i*2 + 1].base = element->data.scalar;
+                         elems[i*2 + 1].len = strlen(element->data.scalar);
+                         h2o_strtolower(elems[i*2 + 1].base, elems[i*2 + 1].len);
+                     }
+                     http2_origin_frame = h2o_mem_alloc(sizeof(*http2_origin_frame));
+                     *http2_origin_frame = h2o_concat_list(NULL, elems, value->data.sequence.size * 2);
```

Comment:
How about creating a function that takes a list of strings to build the frame, instead of having duplicate code for `YOML_TYPE_SCALAR` and `YOML_TYPE_SEQUENCE`?